acme/acme/messages.py

```diff
@@ -328,12 +331,22 @@ class Registration(ResourceBody):
     @classmethod
     def from_data(cls, phone=None, email=None, external_account_binding=None, **kwargs):
         """Create registration resource from contact details."""
+        # extra variable to detect non-null empty lists or tuples
+        details_provided = 'contact' in kwargs and kwargs['contact'] is not None
```

Comment:
I don't `contact` should ever be `None`. If it is and `phone` or `email` is provided, we'll call `None.append` which won't go well.
```suggestion
        details_provided = 'contact' in kwargs
```

Reply:
I don't think that's right, because we provide a default to `kwargs.pop` on line 337 (and did in the previous version of the code), so we don't risk trying to call methods on None.

Reply:
Well I was wrong about the place we blow up:
```
In [1]: kwargs = {'contact': None}

In [2]: details = list(kwargs.pop('contact', ()))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-b877192e83a3> in <module>
----> 1 details = list(kwargs.pop('contact', ()))

TypeError: 'NoneType' object is not iterable
```
`kwargs.pop('contact', ())` will return `None` if `contact` is in `kwargs` and set to `None`. It will only return an empty tuple if it is not provided at all.

Reply:
Oops, I forgot that defaults are used when the key isn't defined, not when the key is `None`. Some experienced python dev I am 😳